### DISCUSSION TITLE
Playlists duplicated in Playlist page

### DISCUSSION THREAD

**Discussion by ceejayemm** [rern/rAudio#986] [Fixed Issues]
rAudio-1: 22020813 64 Bit
RPi: 3A
Shared Data: Enabled

When I create a new Playlist it is being duplicated on the Playlist page:
![ksnip_20220813-150924](https://user-images.githubusercontent.com/25418025/184497960-0a2c8afb-bee9-4ce7-a8cf-4a7a2c93fa38.png)

 but not on the file system:
![ksnip_20220813-150856](https://user-images.githubusercontent.com/25418025/184497947-f3ee39cb-2dee-4e4b-a1f9-4b4e1bbd688e.png)

I haven't really used saved Playlists before so don't know if this is a new problem or one which has been about for a while.  As noted above I have Shared Data enabled and the new playlists are being created correctly on the Shared Data area as seen in the first image above.

Regards.

Chris

**Comment by ceejayemm** [comment]
This is still an issue with 20220828, any chance of a fix ?

Thanks

Chris

**Reply by rern** [reply]
rAudio gets saved playlists by: `mpc lsplaylists`
What is the output of the command on your system?

**Comment by ceejayemm** [comment]
@rern 

Sorry for the delay in replying, have been away from home for a few days.

The above command returns the following:

rAudio-1:~  #  mpc lsplaylists
ABBA
Random Christmas Tracks
Chris Rea
ABBA
Random Christmas Tracks
Chris Rea

Please note this system has Shared Data enabled and is the only one (of 3 system sharing data) which exhibits this problem. The same command on my other (two, which share the same data) systems returns:

rAudio:~  #  mpc lsplaylists
ABBA
Random Christmas Tracks
Chris Rea

Thanks

Chris

**Comment by ceejayemm** [comment]
Still an issue in 20220917.

Chris

**Reply by rern** [reply]
Please try:
- Disable Shared Data on the one with the issue
- Verify that `/srv/http/data/playlists` is empty
- Enable Shared Data again

If still no:
- Remove all playlists
- Create a new playlist and verify on each system

**Comment by ceejayemm** [comment]
@rern 
I think I have solved this problem.  Your information above helped to some extent.  It turned out that I had two 'problems'.

Problem 1:
One of the 'client'  RPi had the /etc/mpd.conf file containing the setting:
playlist_directory     "/var/lib/mpdplaylists"
instead of 
playlist_directory     "/srv/http/data/playlists"

correcting this made both 'client' RPi's display the same set of duplicates.

Problem 2:
Although I use the standard 'SharedData' option, I had taken it a stage further to implement a MPD Satellite sharing system (following https://wiki.archlinux.org/title/Music_Player_Daemon/Tips_and_tricks#Music_streaming_with_the_satellite_setup)

The RPi without the duplicates was the 'server' system whilst the duplicates were being displayed on both the two 'client' systems. It seems to be that rAudio-1 stores its Playlists in the "/srv/http/data/playlists" folder AND in the database as well resulting in the duplicated Playlists being displayed. Reverting to the 'standard' SharedData setup removed the 'duplicate' entries in the display.

The benefit of the MPD Satellite setup (to me) is that albums added to the server system are immediately shared to the other client systems without the need for a database update on the clients. However, with the current setup, it results in apparent duplicate playlists. I will now need to decide if the benefits (to me) of the Satellite setup outweigh the use of the 'standard' SharedData' setup which needs a database update on both clients to make new albums accessible on these system, but with no duplicate playlists. I could simply turn off SharedData and implement my own custom satellite setup but I would then lose the benefits of SharedData in terms of Bookmarks, WebRadios, lyrics etc which do make life a lot easier for me when using multiple rAudio-1 installations.

Would you consider modifying the current SharedData setup to take on board the MPD Satellite concepts ? I think the satellite setup is actually a better solution to some aspects of the current SharedData setup and wouldn't be that much more difficult to implement - if the problem of duplicate playlists can be resolved - letting the database, rather than the file system, deal with playlists might be the obvious answer.

Many thanks for your insights into the workings of the current playlists which have, to some extent, helped me (kind of) sort out this problem.

Please keep up your great work on this project.

Regards.

Chris

**Reply by rern** [reply]
Hi Chris,

A restart of `mpd` will get the latest database from Shared Data.

Please try the revised UPDATE branch which:
- Add IP to Shared Data IP list on power on/boot (previously rely on fixed IPs)
- Remove IP from the list on power off/reboot
- On Library database update - Verify existing of each IP in the list and broadcast command to restart `mpd`

**Comment by ceejayemm** [comment]
@rern 
I removed all my 'satellite' configuration from all my RPi's and returned them to thier original state.  I then applied the UPDATE branch as requested above.  My issues with duplicated playlists do seem to be resolved.

However, when I add a new album to one system and running a database update (from the Cog menu / Library / Changed Files only) the album info is added to the library but the tracks are NOT added until I run a database update from the NAS / Update Database option.  I then still have to run the Update Thumbnails to add the art work.  So nearly there but not quite.  

What is the difference between updating the database from the 'Cog menu / Library / Changed Files' only and from the 'NAS / Update Database' options ? Can these options be combined and, preferably, include the Thumbnail update as well ?  This would make life so much easier.

Thanks for the quick response.

Chris
